src/main/java/io/openliberty/tools/common/plugins/util/ServerFeatureUtil.java

```diff
@@ -380,18 +382,46 @@ private Properties getBootstrapProperties(File bootstrapProperties) {
     private String evaluateExpression(Properties properties, String expression) {
         String value = expression;
         if (expression != null) {
-            Pattern p = Pattern.compile("\\$\\{([^\\}]*)\\}");
+            Pattern p = Pattern.compile("\\$\\{(.*?)\\}");
             Matcher m = p.matcher(expression);
             StringBuffer sb = new StringBuffer();
             while (m.find()) {
                 String variable = m.group(1);
                 String propertyValue = properties.getProperty(variable, "\\$\\{" + variable + "\\}");
+                
+                // Remove encapsulating ${} characters and validate that a valid liberty directory property was configured
+                propertyValue = removeEncapsulatingEnvVarSyntax(propertyValue, properties); 
+
                 m.appendReplacement(sb, propertyValue);
             }
             m.appendTail(sb);
             value = sb.toString();
         }
         return value;
     }
-    
+
+    private String removeEncapsulatingEnvVarSyntax(String propertyValue, Properties properties){
+        Pattern p = Pattern.compile("\\$\\{(.*?)\\}");
+        Matcher m = p.matcher(propertyValue);
+        StringBuffer sb = new StringBuffer();
+        while (m.find()) {
+            String envDirectoryProperty = m.group(1);
+            if(!VALID_LIBERTY_DIRECTORY_PROPERTY.contains(envDirectoryProperty)) {
+                // Check if property is a reference to a configured bootstrap property
+                String bootStrapValue = properties.getProperty(envDirectoryProperty, "\\$\\{" + envDirectoryProperty + "\\}");
+                if(bootStrapValue != null) {
+                    m.appendReplacement(sb, removeEncapsulatingEnvVarSyntax(bootStrapValue, properties));
+                }
+                else {
+                    warn("The directory property " + envDirectoryProperty + "specified is not a predifined Liberty directory property or a configured boostrap property.");
```

Comment:
I would take out the first reference to directory in this log message....change to 

"The referenced property " +envDirectoryProperty + "is not a predefined Liberty directory property or a configured bootstrap property."